Web/Edubase.Web.UI/Areas/Groups/Views/Group/EditLinks.cshtml

```diff
@@ -128,7 +128,7 @@
                     <div class="button-row govuk-!-margin-bottom-4">
                         @if (!Model.LinkedEstablishments.Establishments.Any(x => x.EditMode))
                         {
-                            <button type="submit" name="action" class="govuk-button" value="@VM.ActionSaveLinks">Save @Model.ListOfEstablishmentsPluralName.ToLower()</button>
+                            <button id="save-@StringUtils.ElementIdFormat(@Model.ListOfEstablishmentsPluralName)-top-button" type="submit" name="action" class="govuk-button" value="@VM.ActionSaveLinks">Save @Model.ListOfEstablishmentsPluralName.ToLower()</button>
```

Comment:
A copy/pasted of my comment on the other PR which looks like it might apply here if it's using the establishment name.

Note that establishment names can contain additional characters which are invalid for HTML ID values -- URN 155806 on our test environment is a particularly "special" example.




----


It would be worth double checking if this implementation is sufficient to meet this need, and would be robust enough. 

My thought process (and current assumption) is that `@name` refers to a file name or perhaps a display name?

If so, how does it handle apostrophes like for Children's Centres (noting that apostrophes are not valid characters for an ID value):

![image](https://user-images.githubusercontent.com/96429508/213682025-cd10ec51-6135-4f61-bd5d-ab4228b1fee0.png)

```csharp
// Current implementation:

namespace Edubase.Web.UI.Helpers
{
    public class StringUtils
    {
        public static string ElementIdFormat(string text)
        {
            return text.Replace(' ', '-').ToLowerInvariant();
        }
    }
}
```

```csharp
// Suggested implementation

using System;
using System.Text.RegularExpressions;
using Edubase.Web.UI.Helpers;


public class Program
{
    public static void Main()
    {
        Console.WriteLine("Hello World");
        Console.WriteLine(StringUtils.ElementIdFormat("example with spaces"));
        Console.WriteLine(StringUtils.ElementIdFormat("example with ' apostrophes"));
        Console.WriteLine(StringUtils.ElementIdFormat("example with \" quotes"));
        Console.WriteLine(StringUtils.ElementIdFormat("example with [] square brackets"));
        Console.WriteLine(StringUtils.ElementIdFormat("example with () round brackets"));
        Console.WriteLine(StringUtils.ElementIdFormat("example with accents -- The most common accents are the acute (é), grave (è), circumflex (â, î or ô), tilde (ñ), umlaut and dieresis (ü or ï – the same symbol is used for two different purposes), and cedilla (ç)."));
    }
}


namespace Edubase.Web.UI.Helpers
{
    public class StringUtils
    {
        public static string ElementIdFormat(string text)
        {
            var pattern = @"[^a-zA-Z0-9_]";
            var replacement = "-";
            var result = Regex.Replace(text, pattern, replacement);

            return result.ToLowerInvariant();
        }
    }
}
```


https://dotnetfiddle.net/X4b31o

![image](https://user-images.githubusercontent.com/96429508/213687162-77bd716b-0a1d-437f-80d6-90dd8cb55a75.png)

Reply:
Added your changes in with the extra string.empty replacements.